src/foam/nanos/dig/HTTPDigestSink.java

```diff
@@ -93,6 +94,15 @@ public void put(Object obj, Detachable sub) {
       String payload = outputter.stringify(fobj);
       String digest = getDigest(getX(), dugDigestConfig_, payload);
       conn.addRequestProperty("payload-digest", digest);
+
+      if ( fobj instanceof ExternalDataAware ) {
+        ExternalDataAware extDataObj = (ExternalDataAware) fobj;
+        HashMap<String, String> externalData = (HashMap<String, String>) extDataObj.getExternalData();
+        if ( externalData.containsKey("tid") ) {
+          conn.addRequestProperty("intuit_tid", externalData.get("tid"));
+        }
+      }
+
```

Comment:
This doesn't belong in here as it's Intuit specific code. You'll need to change this to a method call like this: 
```suggestion
    setCustomConnectionProperties(x, fobj, conn);
```
and then in this class the setCustomConnectionProperties method should be empty.
In another inheriting class - e..g IntuitHttpDigestSink - you should override this method and add this code. This sink code should be in the partner/intuit directory.